apps/worker/src/app/workflow/usecases/send-message/digest/digest.usecase.ts

```diff
@@ -67,19 +67,29 @@ export class Digest extends SendMessageType {
     );
   }
 
-  private async getEvents(command: SendMessageCommand) {
-    const currentJob = await this.jobRepository.findOne({ _environmentId: command.environmentId, _id: command.jobId });
+  private async newGetEvents(command: SendMessageCommand) {
+    const currentJob = await this.getCurrentJob(command);
 
-    if (!currentJob) {
-      const message = `Digest job ${command.jobId} is not found`;
-      Logger.error(message, LOG_CONTEXT);
-      throw new PlatformException(message);
-    }
+    const jobs = await this.jobRepository.find(
+      {
+        _mergedDigestId: currentJob._id,
+        status: JobStatusEnum.MERGED,
+        type: StepTypeEnum.DIGEST,
+        _environmentId: currentJob._environmentId,
+        _subscriberId: command._subscriberId,
+      },
+      'payload'
+    );
+
+    return [currentJob.payload, ...jobs.map((job) => job.payload)];
+  }
+
+  private async getEvents(command: SendMessageCommand) {
```

Comment:
Would suggest to call this one:

```suggestion
  private async backwardCompatibleGetEvents(command: SendMessageCommand) {
```

And add a comment, and for the new one just calling getEvents